### ISSUE TITLE
Cron tasks do not follow the timing rules

### ISSUE THREAD

**Issue by f1sle** [strapi/strapi#12075]
### Describe the bug
Cron tasks do not follow the timing rules and start every minute.

### Steps to reproduce the behavior

1. I create a file 'cron-tasks.js' in a config folder of a project with the following contents:
`const checkForNewReviews = require("./functions/checkForNewReviews");

module.exports = {
  // Check new reviews every day at midnight
  "0 0 * * *": {
    task: ({ strapi }) => {
      checkForNewReviews();
    },
    options: {
      tz: "America/Los_Angeles",
    },
  },
};`

2. A trigger function was created '**checkForNewReviews.js**' `/config/functions/checkForNewReviews.js` with the following contents:
`require("dotenv").config();
const axios = require("axios");

module.exports = async () => {
  try {
    await axios({
      url: `http://${process.env.HOST}:${process.env.PORT}/api/reviews/update`,
      method: "post",
      responseType: "json",
      headers: {
        Authorization: `Bearer ${process.env.API_SECRET_TOKEN}`,
        "Content-Type": "application/json",
      },
    });
  } catch (error) {
    console.warn(error);
  }
};`

3. The problem occurred in a dev env as well as in prod, and every time POST request was triggered every minute.

### Expected behavior
Cron task should run according to the time rules of a cron-tasks.js file.

### Code snippets
/config/functions/checkForNewReviews.js
![image](https://user-images.githubusercontent.com/39921649/147888105-1fc38077-8f1b-44c9-bd0f-8d1954eb6998.png)

/config/cron-tasks.js
![image](https://user-images.githubusercontent.com/39921649/147888122-23ea791a-1f67-438e-a277-26e357775fc3.png)

Console.log of a prod build:
![image](https://user-images.githubusercontent.com/39921649/147888142-caa71e38-4e50-411c-b8aa-4c0f99501539.png)

If applicable, add code samples to help explain your problem.

### System

- Node.js version: v14.17.6
- NPM version: 7.21.1
- Strapi version: 4.0.0
- Database: Postgresql 14
- Operating system: macOS
.

**Comment by f1sle** [comment]
When I'm trying to schedule it for every 10 seconds, it fails, and instead of it runs every minute.
Tried both ways: */10 * * * * * and 10 * * * * * .

**Comment by derrickmehaffy** [comment]
`0 0 * * *` is not every every minute, that would be `* * * * *`

The cron system comes from https://www.npmjs.com/package/node-cron we don't modify it past that, so if there is an issue going below a minute (default cron structure doesn't allow crons below a minute) then it would be a problem with node-cron not us.

**Comment by f1sle** [comment]
> 

Yeah, 0 0 * * * it's not every minute, it's at midnight every day, but it runs every minute...

**Comment by HannesOberreiter** [comment]
Did run into the same problem, with Docker `strapi/base:latest` + Strapi V4. Task runs every minute doesn't matter what I write. 

Cheers
Hannes

**Comment by signly-projects** [comment]
Same here :+1: 
Runs every minute no matter what cron format I use.

**Comment by derrickmehaffy** [comment]
I spent probably far too long seeing if I could reproduce this using about 50 different cron strings.

First thing I'll preface this by saying you really shouldn't be building crons by the second or anything less than 60 seconds because it's not even part of the cron specification, it's isolated to this specific package.

If you want to see how or when a **normal** cron string will run I advise you check out https://crontab.guru

---

Now that being said, I cannot reproduce on any of the following operating systems (of which I've tried it on all of them with this exact same code):

- Linux Mint 20.3
- Ubuntu 20.10 Server
- Ubuntu 20.04 Server
- CentOS Stream
- Manjaro 21.2.2
- Windows 11
- Windows 10
- Windows 7
- MacOS Big Sur
- MacOS Catalina
- Docker (Ubuntu 20.04 focal)
- Docker (Ubuntu 18.04 bionic)
- LXD (Various Linux OSes)

![image](https://user-images.githubusercontent.com/8593673/151455755-54c9531e-07ff-4120-a89a-20d4f7789eca.png)

If there is something I'm missing or you have some very odd environment that I haven't tested I would love to see it

**Comment by f1sle** [comment]
@derrickmehaffy, I guess it was a documentation issue, I'll give it a try later today, but I think everything will be fine, coz I see that you did that in a 'server.js' file... not 'cron.js'

**Comment by derrickmehaffy** [comment]
Doesn't matter which file you do it in, the file can be anything you want and you simply require it and set it on the tasks key.

I helped write that documentation :) 
https://github.com/strapi/documentation/pull/463

**Comment by f1sle** [comment]
@derrickmehaffy, Well, I figured out what caused that odd behavior.
If you add timezone option into 'cron-tasks.js' file:

`const checkForNewReviews = require("./functions/checkForNewReviews");

module.exports = {
  // Check new reviews every day at midnight
  "0 0 * * *": {
    task: ({ strapi }) => {
      checkForNewReviews();
    },
    //options: {
    //  tz: "America/Los_Angeles",
    //},
  },
};`

then it will execute your tasks every minute...

**Comment by HannesOberreiter** [comment]
Thanks for your great effort. As @f1sle stated its running node-schedule and not node-cron (https://github.com/strapi/strapi/blob/master/packages/core/strapi/package.json). Also great find by @signly-projects that the options seems to break it, though I tested it without and with but seems not like it.

Following setup now works for me:

```javascript
module.exports = {
    'rss': {
       task: async ({ _strapi }) => { 
           console.log("I am running every 2 minutes >> " + new Date());
       },
       options: {
           rule: '*/2 * * * *',
           tz: 'Europe/Vienna',
       },
    }
};
```

Cheers
Hannes

**Comment by derrickmehaffy** [comment]
Looks like we swapped out libraries at some point that got missed, I believe most of the docs should still work but maybe not all the keys are correct.

I'll transfer this to the documentation repo and if anyone feels like making a PR after you do some more advanced testing please do.
